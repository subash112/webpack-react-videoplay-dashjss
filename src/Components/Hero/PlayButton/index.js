import React from 'react'

const PlayButton = () => {
    return(
        <p>play</p>
    )
}

export default PlayButton