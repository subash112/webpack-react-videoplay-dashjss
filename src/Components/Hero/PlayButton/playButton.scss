.play-button{
    padding: 2rem;
    background-color: #171717;
    border-radius: 10rem;
    height: 90;
    width: 100;
}