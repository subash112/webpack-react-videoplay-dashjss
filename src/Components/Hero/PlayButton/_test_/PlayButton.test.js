import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import PlayButton from '../index';
import { BrowserRouter } from 'react-router-dom';

afterEach(cleanup);
it('renders without crashing', () => {
  const div = document.createElement('div');

  ReactDOM.render(
    <BrowserRouter>
      <PlayButton />
    </BrowserRouter>, 
  div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
    const {getByTestId} = render(
        <BrowserRouter>
          <PlayButton id = "Movie-to-test"/>
        </BrowserRouter>
      )
    expect(getByTestId("PlayButton").getAttribute("href")).toBe(
        "/stream/movie/Movie-to-test")

})

it('matches snapshot', () => {
    const tree = TestRenderer.create(
    <BrowserRouter>
        <PlayButton id = "Movie-to-test"/>
    </BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
})
