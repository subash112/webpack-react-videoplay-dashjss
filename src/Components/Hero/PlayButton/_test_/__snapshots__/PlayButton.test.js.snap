// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="play-button"
>
  <a
    data-testid="PlayButton"
    href="/stream/movie/Movie-to-test"
    onClick={[Function]}
  >
    <img
      alt="play icon"
      height="100"
      src="test-file-stub"
      width="100"
    />
  </a>
</div>
`;
